freqtrade/commands/data_commands.py

```diff
@@ -43,16 +43,18 @@ def start_download_data(args: Dict[str, Any]) -> None:
         if config.get('download_trades'):
             pairs_not_available = refresh_backtest_trades_data(
                 exchange, pairs=config["pairs"], datadir=config['datadir'],
-                timerange=timerange, erase=config.get("erase"))
+                timerange=timerange, erase=cast(bool, config.get("erase")))
```

Comment:
Any particular reason / advantage to use `cast(bool, xxx)` instead of `bool(xxx)`?

we're not using cast this way anywhere else ...

Reply:
Performance. `cast` generates no byte-code.

Reply:
Why would you opt for the slower method? :laughing: 

![2020-02-03-071939_653x157_scrot](https://user-images.githubusercontent.com/5024695/73630176-a2759200-4655-11ea-8315-2d606a88036e.png)

![2020-02-03-072006_674x168_scrot](https://user-images.githubusercontent.com/5024695/73630182-a7d2dc80-4655-11ea-857d-76b8bc622f44.png)

Reply:
These are all your notebooks...

```
$ python3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from timeit import timeit
>>> timeit("cast(bool, 200)", "from typing import cast")
0.11175005696713924
>>> timeit("bool(200)")
0.12564564507920295
>>> timeit("cast(bool, '200')", "from typing import cast")
0.11431943299248815
>>> timeit("bool('200')")
0.1346755709964782
>>> 
```

Reply:
> we're not using cast this way anywhere else ...

Well, we probably should in the future, in all other places where `float()`, `str()`, etc are called just to make mypy/flake happy...

Reply:
Wrong, it's the python version that matters, notebook or not does not matter for execution speed.

`bool()` is faster in 3.8 - and i'd stick to that since that's the "more modern".
3.7 is faster than 3.6, 3.8 is supposed to be faster than 3.7 ... 

```
python
Python 3.8.0 (default, Oct 18 2019, 09:56:01)
[GCC 7.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from timeit import timeit
>>> timeit("cast(bool, 200)", "from typing import cast")
0.09076810000010482
>>> timeit("bool(200)")
0.07849940000005518
>>> timeit("cast(bool, '200')", "from typing import cast")
0.1061870999999428
>>> timeit("bool('200')")
0.07970810000006168
>>>
```

Reply:
* Your examples ^^ are quite incorrect, since it's not really type conversion, the value is already bool, so should better testes as `cast(bool, True)` and `bool(True)` (and `bool('<int>')` always returns True, even for `bool('0')` 😉 )

* Okay, I see the same: in python 3.7 ans python 3.7 `bool()` is faster than `cast()` (as well as calls to `float()` and `int()` are faster as well). Which is strange because `cast` assumes to 'return the value as it is', it's simply a hint for the static checker that the developer knows the value *is to be* of expected type... And `bool()` conversion should at least perform type checking to see that the value is already bool and should simply be returned without conversion... So it's strange a bit...

* As I said at the top, this is more about how the code will look in the future, when similar recipes are applied in the codebase. We do not use `cast`, right, and I can simply change it here to `bool`, as in other places, we'll be always able to refactor code in the codebase in the future, But in my opinion `cast` is more ideomatic (however, lots of legacy python code uses type conversion functions -- that's because python typing is relatively new in many of its parts) because it simply hints static checker that the untyped value is of expected type. The conversion functions can modify the values if not used properly, like in the example of `bool('0')` above...

Summary: changing it to `bool`, right?

Reply:
Very strange why cast is slower:
https://github.com/python/cpython/blob/4b524161a0f9d50d782e739a3708434ffd4e94a5/Lib/typing.py#L1146-L1154

Reply:
> Summary: changing it to bool, right?

yes
using cast() with the link you provided is a dangerous thing - it has you assume that a type is converted - even though it's not.

I'm also comming from another language background (not c/cpp) - but when using cast, i always assume that the returned type is of the new type - and not just telling linters / typecheckers it's that way.
So using `cast(float, '200')` is even worse than slower - it's giving us a completely false sense of correctness as you think "well - i did cast it" - but the return type is still a string (`cast(float, '200') == '200'`).

> Very strange why cast is slower

While this only returns the called function - it's a function call nonetheless.
I have to assume that `bool()` is some very heavily optimized piece of C-code (it's one of the core-types) ... even though i'm too lazy (and not interrested enough) in finding it in the Cpython codebase.

Reply:
> it has you assume that a type is converted - even though it's not.

No, using cast says to the interpreter (well, actually to the static type checker) that 'developer knows (or assumes) that the value is already of this type and no conversion required'. That's the semantics of the typing.cast.

Reply:
of `typing.cast` - yes - which i find highly confusing (but i'm sceptical of everything i import from typing, anyway) - but in c# using cast (syntax is different though) does actually convert the type ...

Reply:
I'm only talking about python cast here. Using terminology from other languages, yes, I would say that `bool()` and `float()` are real casts, exactly... 🙄 
All above, where I use the term 'cast' -- that's about python typing.cast.

Reply:
i know ... :+1: